### PR TITLE
2022 Markup Chapter

### LINKED ISSUE #2881: Markup 2022
# Markup 2022

Markup illustration

If you're interested in contributing to the Markup chapter of the 2022 Web Almanac, please reply to this issue and indicate which role or roles best fit your interest and availability: [author](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide), [reviewer](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide), [analyst](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide), and/or [editor](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide).

## Content team

| Lead | Authors | Reviewers | Analysts | Editors | Coordinator |
| ---- | ------- | --------- | -------- | ------- | ----------- |
| ~@AlexLakatos~ @j9t | ~@AlexLakatos @ibnesayeed @daKmoR~ @j9t | ~@j9t~ @zcorpan @bkardell | @rviscomi | - | @siakaramalegos |

<details>
<summary><b>Expand for more information about each role 👀</b></summary>

- The **[content team lead](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Content-Team-Leads'-Guide)** is the chapter owner and responsible for setting the scope of the chapter and managing contributors' day-to-day progress.
- **[Authors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide)** are subject matter experts and lead the content direction for each chapter. Chapters typically have one or two authors. Authors are responsible for planning the outline of the chapter, analyzing stats and trends, and writing the annual report.
- **[Reviewers](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide)** are also subject matter experts and assist authors with technical reviews during the planning, analyzing, and writing phases.
- **[Analysts](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide)** are responsible for researching the stats and trends used throughout the Almanac. Analysts work closely with authors and reviewers during the planning phase to give direction on the types of stats that are possible from the dataset, and during the analyzing/writing phases to ensure that the stats are used correctly.
- **[Editors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide)** are technical writers who have a penchant for both technical and non-technical content correctness. Editors have a mastery of the English language and work closely with authors to help wordsmith content and ensure that everything fits together as a cohesive unit.
- The **[section coordinator](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Section-Leads'-Guide)** is the overall owner for all chapters within a section like "User Experience" or "Page Content" and helps to keep each chapter on schedule.

_Note: The time commitment for each role varies by the chapter's scope and complexity as well as the number of contributors._

For an overview of how the roles work together at each phase of the project, see the [Chapter Lifecycle](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle) doc.
</details>

### [**0. Form the content team**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#0-create-content-team)
+ [x] **May 1**: The content team has at least one author, reviewer, and analyst

### [**1. Plan content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#1-plan-content)
+ [x] **May 15** The content team has completed the chapter outline in the [draft doc][~google-doc]

### [**2. Gather data**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#2-gather-data)
+ [x] **June 1**: Analysts have added all necessary [custom metrics](https://github.com/HTTPArchive/custom-metrics/blob/main/README.md) and drafted a PR ([example](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1087)) to track query progress
- **June 1 - 15**: HTTP Archive runs the June crawl

### [**3. Validate results**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#3-validate-results)
+ [x] **August 1**: Analysts have queried all metrics and saved the output to the [results sheet][~google-sheets]

### [**4. Draft content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#4-draft-content)
+ [x] **September 1**: The content team has written, reviewed, and edited the chapter in the [doc][~google-doc]

### [**5. Publication**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#5-publication)
+ [ ] **September 15**: The completed chapter and all required [metadata](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters) and [figures](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide) are converted to markdown and submitted to [GitHub][~chapter-markdown]
- **September 26**: Target launch date 🚀

## Chapter resources

Refer to these 2022 Markup resources throughout the content creation process:

[📄 Google Docs][~google-doc] for outlining and drafting content
[🔍 SQL files][~sql-files] for committing the queries used during analysis
[📊 Google Sheets][~google-sheets] for saving the results of queries
[📝 Markdown file][~chapter-markdown] for publishing content and managing public metadata
[💬 #web-almanac-markup on Slack][~slack-channel] for team coordination

[~google-doc]: https://docs.google.com/document/d/1OV_-YmxXoFCalDIdAYq9eTZMpjsQ0rsuFKGAsF-8eoM/edit?usp=sharing
[~sql-files]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/sql/2022/markup
[~google-sheets]: https://docs.google.com/spreadsheets/d/1grkd2_1xSV3jvNK6ucRQ0OL1HmGTsScHuwA8GZuRLHU/edit?usp=sharing
[~chapter-markdown]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content/en/2022/markup.md
[~slack-channel]: https://join.slack.com/t/httparchive/shared_invite/zt-45sgwmnb-eDEatOhqssqNAKxxOSLAaA

### PR DESCRIPTION
**Pull request by tunetheweb**
Fixes #2881 

Staging URL: https://20220923t160735-dot-webalmanac.uk.r.appspot.com/en/2022/markup
Home page quotes: https://20220923t160735-dot-webalmanac.uk.r.appspot.com/en/2022/?feat=markup#featured-chapter

This includes the edits and conversion of the [Markdown Doc](https://docs.google.com/document/d/1OV_-YmxXoFCalDIdAYq9eTZMpjsQ0rsuFKGAsF-8eoM/edit#). I left the edits in the doc so you can see what I changed, but incorporated them into this markdown conversion under the hope none of them are too controversial. Please comment on this PR rather than the doc with any suggestions.

- [Doc](https://docs.google.com/document/d/1OV_-YmxXoFCalDIdAYq9eTZMpjsQ0rsuFKGAsF-8eoM/edit#)
- [Sheet](https://docs.google.com/spreadsheets/d/1grkd2_1xSV3jvNK6ucRQ0OL1HmGTsScHuwA8GZuRLHU/edit#gid=1535288056)

### CODE REVIEW

File: src/content/en/2022/markup.md
Comment (tunetheweb) [inline src/content/en/2022/markup.md:7]: @j9t can you confirm these contributors? I remove the others that I don't believe contributed but let me know if I've wrong.

I'm not sure if anyone other than myself reviewed your doc?
@bkardell you were down as a reviewer, but not sure if you did review or not? Know you're busy with your own chapter, but if you do want to review you can see the doc here: https://20220923t160735-dot-webalmanac.uk.r.appspot.com/en/2022/markup

File: src/content/en/2022/markup.md
Comment (tunetheweb) [inline src/content/en/2022/markup.md:10]: @j9t I copied this from your previous bio. Feel free to update as you see fit.

File: src/content/en/2022/markup.md
Comment (tunetheweb) [inline src/content/en/2022/markup.md:18]: @j9t I picked a quote and a few stats I thought might fit, but normally author's pick these so feel free to change as you see fit. [Here's what it looks like](https://20220923t160735-dot-webalmanac.uk.r.appspot.com/en/2022/?feat=markup#featured-chapter) with these:

image

**Comment by tunetheweb** [comment]
And you'll be please to know we do validate our HTML (and run Lighthouse tests on it too!) and, apart from the non-standard `fetchpriority` attribute, we're all clean!

File: src/content/en/2022/markup.md
Comment (rviscomi) [reply src/content/en/2022/markup.md:7]: @zcorpan also [gave it a look](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2881#issuecomment-1238156094) and should get reviewer credit. Could also be good to give it another look in the website preview and leave any additional feedback.

**Comment by j9t** [comment]
Thanks @tunetheweb! Will review asap (likely tomorrow).

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md]: Ah missed that. Good so @zcorpan and @bkardell reviewed. I'll remove my name and settle for editor. Just didn't want to give the impression it was not given the appropriate scrutiny, but seems it was!

**Review by j9t — CHANGES_REQUESTED** [review]
@tunetheweb, thank you!

I focused only on the PR now, so let me know if there’s also something in the doc that I should review. I saw there were comments and suggestions.

For transparency around any changes on my end, I made only a few changes myself (case in table heads, for consistency; some wording swap, noted in the PR; updated bio). I didn’t notice anything major, so the things commented here should be fairly straightforward.

From my end, all good then. Does it work for you and team if we eventually make some small tweaks later, too? I just want to make sure that I didn't miss anything, and do a final review once we published the chapter. (Open to other thoughts, if the published version should not be tweaked again.)

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:4]:
```suggestion
description: Markup chapter of the 2022 Web Almanac covering document data (doctypes, compression, languages, HTML conformance, document size), the use of HTML elements and attributes, data attributes and social media.
```

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:107]: Yesh Web Almanac is American English.  But it's difficult for me to override 44 years of learning :-)

```suggestion
Are HTML documents being compressed? How many? How? 86% of them are—with 58% (down 5.8% since last year) overall being gzip-compressed, and 28% (up 6.1%) being compressed using Brotli. To summarize, slightly more documents are being compressed, and compressed more effectively.
```

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:122]:
```suggestion
What about languages? In our data set, 35% of pages used a lang attribute mapping to English; 17% had no language set; and you already see the difficulties—the sample is likely biased and also not as big as to reflect all of the world, and no `lang` attribute being used is not equaling no language being set so, this isn't something our data would be useful for.
```

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:151]: That's the title of the post with capital N as CCS Tricks uses Title Case. However should also be capital W then.

```suggestion
Perhaps if we wrote HTML <a hreflang="en" href="https://css-tricks.com/write-html-the-html-way-not-the-xhtml-way/">the HTML Way (Not the XHTML Way)</a> that would result in smaller HTML transfer size, and would counter this trend. _Disclosure: Your author here likes to come up with HTML writing classifications, and enjoys promoting minimal HTML._
```

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:300]:
```suggestion
But here it's not a surprise that not every document uses `html`, `head`, or `body` tags—but not because the respective tags are optional in HTML. As they are automatically inserted in the DOM, they are being counted here. That the numbers are slightly less than 100% is due to a small number of pages that break detection by overriding the JavaScript APIs we use—for example, <a hreflang="en" href="https://mootools.net/">MooTools</a> overriding the `JSON.stringify()` API.
```

Comment (tunetheweb) [reply src/content/en/2022/markup.md:300]: Got another suggestion above. Let me know what you think.

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:302]:
```suggestion
It's a lot more surprising to miss `title` on 1% of all sampled documents—this element is not optional, and not being inserted in the DOM, and its omission an indicator for lack of conformance checking.
```

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:386]:
```suggestion
```

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:467]:
```suggestion
The most popular attribute, by far, was and still is `class`:
```

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:550]: Personally I don't see need for comma here.

I wouldn't say: "The number of elements per page goes up every year <pause> too"

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:554]: I suggest removing this comma as doesn't seem necessary to me:

```suggestion
What perhaps we'll also be able to do is to look at conformance too. Not everyone may care about this at this time in our field. But we're all professionals, and it seems at least relevant to know whether overall, we produce work that corresponds to <a hreflang="en" href="https://html.spec.whatwg.org/multipage/">the underlying standard(s)</a>. After all, this shouldn't be a chapter about fantasy HTML—it should be one about HTML that actually works. It's one of the most important web standards.
```

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:18]: Good idea [concentrating on the positive is one of our principles!](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#principles)

```suggestion
featured_quote: Without HTML there are no web pages, no web sites, no web apps. You can say that without HTML, there's no Web. That makes HTML one of the most important web standards, if not the most important web standard.
featured_stat_1: 90%
featured_stat_label_1: Documents using the HTML5 doctype.
featured_stat_2: 30 KB
featured_stat_label_2: Median HTML document transfer size
featured_stat_3: 29%
featured_stat_label_3: Elements which are `div`s.
```

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:5]: Editor's fine. All part of the service! 😁

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:550]: Take two with the comment above as GitHub swallowed my `<pause>` point!

File: src/content/en/2022/markup.md
Comment (j9t) [reply src/content/en/2022/markup.md:300]: Can we replace “But here it’s” by “It’s”, and in the following paragraph, “It’s a lot more surprising” by “But it is a surprise”? I think that would read better then.

Optional: How about breaking the first sentence up, perhaps by a period after “tags”, and then continuing with “They are…”?

And, what do you think of the double “overriding”? Missed this earlier, but caught it now.

File: src/content/en/2022/markup.md
Comment (j9t) [reply src/content/en/2022/markup.md]: Understand. How about we use the whole title then, and use `<cite>` for it?

If not, this seems to be regular prose—do we have other instances where we use title case?

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:433]: Looked at some sample sites containing it:

- https://www.brickstorino.net/
- https://www.convenioscard.com.br/ 
- https://www.ucenter.org/

Seems it's part of Wix? Which would explain why pretty identical numbers to `wix-images` 

Part of [this file](https://static.parastorage.com/services/wix-thunderbolt/dist/main.b48f6cdc.bundle.min.js)

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md:300]: > Can we replace “But here it’s” by “It’s”, and in the following paragraph, “It’s a lot more surprising” by “But it is a surprise”? I think that would read better then.

Did the first.

Not sure the second is a great idea. Sentences beginning with "But" is generally bad form—it's an indication of a continuation of the previous thought so should really be in the same paragtpah—and for paragraphs even more so!

> Optional: How about breaking the first sentence up, perhaps by a period after “tags”, and then continuing with “They are…”?

I wouldn't break it as seems like the reason it's not a surprise is at least in part because the browser is automatically inserting them. I replaced with an em-dash instead in above suggestion.

I think the double overriding is fine here. We're saying "it's caused by overrides, such as this example override" so it's a deliberate repeat.

Comment (j9t) [reply src/content/en/2022/markup.md:300]: I think it reads better, thanks.

The suggestion to start the next paragraph with “But it is…” was deliberate, for emphasis. (“It is not a surprise that… But it is a surprise that…”.) This is not to argue about it; let’s defer this point and look at how the final/live version reads. Does that work for you?

File: src/content/en/2022/markup.md
Comment (j9t) [reply src/content/en/2022/markup.md]: (Unresolving just to make sure it doesn’t drown. Forgot this earlier.)

File: src/content/en/2022/markup.md
Comment (j9t) [reply src/content/en/2022/markup.md:433]: Ah, awesome. Thanks for researching, too.

Can we rephrase? How about

```
This is more diverse: `pages-css`, `wix-image`, and `wix-iframe` come from the Wix website builder. `router-outlet` originates in Angular. And `ss3-loader` seems to be related to Smart Slider.
```

(Essentially just merging the first two full sentences.)

Comment (tunetheweb) [reply src/content/en/2022/markup.md:433]:
```suggestion
This is more diverse: `pages-css`, `wix-image` and `wix-iframe` come from the Wix website builder. `router-outlet` originates in Angular. And `ss3-loader` seems to be related to Smart Slider.
```

File: src/content/en/2022/markup.md
Comment (tunetheweb) [inline src/content/en/2022/markup.md:151]:
```suggestion
Perhaps if we followed the advice to <cite><a hreflang="en" href="https://css-tricks.com/write-html-the-html-way-not-the-xhtml-way/">Write HTML, the HTML Way (Not the XHTML Way)</a></cite> that would result in smaller HTML transfer size, and would counter this trend. _Disclosure: Your author here likes to come up with HTML writing classifications, and enjoys promoting minimal HTML._
```

File: src/content/en/2022/markup.md
Comment (tunetheweb) [reply src/content/en/2022/markup.md]: Completed in https://github.com/HTTPArchive/almanac.httparchive.org/pull/3147/commits/3cbae21061635b37c711bc5fdb5c3dc779e33693

**Comment by tunetheweb** [comment]
Restaged to 20220923t160735-dot-webalmanac.uk.r.appspot.com/en/2022/markup after your feedback.

> I focused only on the PR now, so let me know if there’s also something in the doc that I should review. I saw there were comments and suggestions.

Nope, as long as you're happy with what's in this PR! That was more for your review in case you were wondering what changed (and in some cases - why).

> Does it work for you and team if we eventually make some small tweaks later, too? I just want to make sure that I didn't miss anything, and do a final review once we published the chapter. (Open to other thoughts, if the published version should not be tweaked again.)

The chapter can be edited after publication—particularly when mistakes are noticed! However, I would caveat that by saving it's not a living document and is supposed to be a snapshot in time. So the intention is not to make material changes to the content (e.g. add completely new sections), nor maintain the data throughout the year.

**Comment by j9t** [comment]
Thanks @tunetheweb!

I commented on [one thing about the CSS-Tricks ref](https://github.com/HTTPArchive/almanac.httparchive.org/commit/3cbae21061635b37c711bc5fdb5c3dc779e33693#r84915250), apart from this this looks great!

> The chapter can be edited after publication—particularly when mistakes are noticed! However, I would caveat that by saving it's not a living document and is supposed to be a snapshot in time. So the intention is not to make material changes to the content (e.g. add completely new sections), nor maintain the data throughout the year.

Makes sense, absolutely. If anything I’d probably prefer one final clean-up PR afterwards, also not to feel bad if anything escapes our attention now.

File: src/content/en/2022/markup.md
Comment (tunetheweb) [inline src/content/en/2022/markup.md:151]:
```suggestion
One way to counter this trend is to <a hreflang="en" href="https://css-tricks.com/write-html-the-html-way-not-the-xhtml-way/">write HTML, the HTML way (and not the XHTML way)</a>, as that would already result in smaller HTML transfer size. _Disclosure: Your author here likes to come up with HTML writing classifications, and enjoys promoting minimal HTML._
```

**Comment by tunetheweb** [comment]
> Makes sense, absolutely. If anything I’d probably prefer one final clean-up PR afterwards, also not to feel bad if anything escapes our attention now.

That's fine but not both Rick and I are away last week, and will likely prioritise any free time we do have, for breaking changes, or to get the other chapters across the line. So just a warning that it won't be instantly reviewed and released.

I committed your last change, so if you're happy to approve (with that caveat), then will merge this once CI completes.

**Comment by j9t** [comment]
> That's fine but not both Rick and I are away last week, and will likely prioritise any free time we do have, for breaking changes, or to get the other chapters across the line. So just a warning that it won't be instantly reviewed and released.

Absolutely, works for me!

> I committed your last change, so if you're happy to approve (with that caveat), then will merge this once CI completes.

Thanks, LGTM!

**Review by j9t — APPROVED** [review]

**Pull request merged**
